source/filtered_table_model.ts

```diff
@@ -0,0 +1,214 @@
+import * as Kola from 'kola-signals';
+import { ArrayTableModel } from './array_table_model';
+import { AddRowOperation, Operation, RemoveRowOperation, UpdateValueOperation }
+  from './operations';
+import { TableModel } from './table_model';
+
+
+export abstract class Predicate {
+
+  /**
+   * @param row - The row to apply the predicate to.
+   */
+  public abstract applyPredicate(row: number, model: TableModel): boolean;
```

Comment:
Make this just a function instead of an abstract class.

Reply:
Should it be a currying function?